crates/ra_editor/src/lib.rs

```diff
@@ -97,13 +106,37 @@ pub fn diagnostics(file: &SourceFileNode) -> Vec<Diagnostic> {
         }
     }
 
-    file.errors()
+    let mut errors: Vec<Diagnostic> = file
+        .errors()
         .into_iter()
         .map(|err| Diagnostic {
             range: location_to_range(err.location()),
             msg: format!("Syntax Error: {}", err),
+            severity: Some(Severity::Error),
         })
-        .collect()
+        .collect();
+
+    let warnings = check_unnecessary_braces_in_use_statement(file);
```

Comment:
Perhaps this could be separated out into `ra_lints` or something (might want to see what @matklad says).

So you'd call `ra_lints::lint_file(file)`.

Reply:
I'd keep this in ra_editor, *for now*. We certainly need to design a better error-reporting infrastructure, but that's a task for the future.